codetransformer/utils/immutable.py

```diff
@@ -178,26 +188,105 @@ def _wrapinit(init):
     wrapped : callable
         The wrapped init method.
     """
-    def __init__(*args, **kwargs):
-        self = args[0]  # they could name this whatever and this breaks kwargs.
-
-        __setattr__._initializing.add(self)
-        init(*args, **kwargs)
-        __setattr__._initializing.remove(self)
-
-        missing_slots = tuple(
-            filter(lambda s: not hasattr(self, s), self.__slots__),
+    try:
+        spec = getfullargspec(init)
+    except TypeError:
+        # we cannot preserve the type signature.
+        def __init__(*args, **kwargs):
```

Comment:
Is there a legitimate value that could be put here that we can't call `getfullargspec` on?